.travis.yml

```diff
@@ -1,21 +1,46 @@
 language: python
-python:
-  - "3.4"
-  - "3.5"
-  - "3.6"
-  - "3.7"
-script:
-  - pytest tests/uts -v
-  - pytest tests/integration -v
+matrix:
+  include:
+    - stage: client unit tests
+      python: 3.4
+      env:
+        - TEST_SUITE="tests/uts/test_client.py"
+    - python: 3.5
+      env:
+        - TEST_SUITE="tests/uts/test_client.py"
+    - python: 3.6
+      env:
+        - TEST_SUITE="tests/uts/test_client.py"
+    - python: 3.7
+      env:
+        - TEST_SUITE="tests/uts/test_client.py"
+    - stage: agent unit tests
+      python: 2.7
+      env:
+        - TEST_SUITE="tests/uts/test_agent.py"
+    - stage: integration tests
+      python: 3.4
+      env:
+        - TEST_SUITE="tests/integration"
+      before_install:
+        - ./travis_before_install.sh
+    - python: 3.5
+      env:
+        - TEST_SUITE="tests/integration"
+      before_install:
+        - ./travis_before_install.sh
+    - python: 3.6
+      env:
+        - TEST_SUITE="tests/integration"
+      before_install:
+        - ./travis_before_install.sh
+    - python: 3.7
+      env:
+        - TEST_SUITE="tests/integration"
+      before_install:
+        - ./travis_before_install.sh
```

Comment:
that is not a very good name for a script file. how about `tests/integration/setup.sh`?